### PR TITLE
Handle missing IdP trait in PAM interpolation.

### LINKED ISSUE #6545: PAM environmental variables break ssh login if corresponding value is NULL
### Description

**What happened**:
Customer enabled PAM on a server and configured a number of environmental variables they expected their IdP to pass back. The IdP was misconfigured, however, and did not pass the expected key:value pairs. This broke all ssh connectivity to the node, including connectivity for existing users, until these environmental variables were removed from the teleport config.

**What you expected to happen**:
If an admin misconfigures their IdP and expected attributes are not passed back to teleport during the authorization process, this should not break all SSH connectivity to the node. We should either ignore the null value or pass in a placeholder if necessary.

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. Set up PAM on a node w/ environmental variables configured in the teleport config.
2. Log in w/ an auth connector but make sure that the expected env variable key:value pair is not passed back by the connector on login.
3. Attempt to SSH to a node via WebUI either with a non-existent or already-existent Linux user.
4. Observe inability to log in.

### Server Details
- Teleport version (run `teleport version`): 6.1.1

### Debug Logs
Please include or attach debug logs, when appropriate. Obfuscate sensitive information!
```
disconnected
ssh: could not start shell
```

### PR DESCRIPTION
**Pull request by xacrimon**
This defaults the environment variable during PAM environment variable interpolation if it does not exist. The fallback value is meant to indicate a misconfiguration to notify admins of the issue.

Needs backport to `branch/v6`.

Fixes #6545

### CODE REVIEW

**Comment by russjones** [comment]
@xacrimon I'm not sure setting a placeholder is the right answer here, what are our alternatives?

**Comment by russjones** [comment]
@fspmarshall @r0mant Can you review?

File: lib/srv/ctx.go [hunk 1]
```diff
@@ -725,6 +725,15 @@ func getPAMConfig(c *ServerContext) (*PAMConfig, error) {
 
 			result, err := expr.Interpolate(traits)
 			if err != nil {
+				// If the trait isn't passed by the IdP due to misconfiguration
+				// we fallback to setting a value which will indicate this.
+				if trace.IsNotFound(err) {
+					trait := expr.Name()
+					log.Warnf("Attempted to interpolate custom PAM environment with external trait %q but received SAML response does not contain claim %q.", trait, trait)
```
Comment (r0mant) [inline lib/srv/ctx.go:732]:
```go
log.Warnf("Attempted ... with external trait %[1]q but received ... does not contain claim %[1]q", expr.Name())
```

will save you 1 line and 1 variable :)

File: lib/srv/ctx.go [hunk 2]
```diff
@@ -725,6 +725,15 @@ func getPAMConfig(c *ServerContext) (*PAMConfig, error) {
...
+					environment[key] = "TELEPORT_DID_NOT_FIND_TRAIT"
```
Comment (r0mant) [inline lib/srv/ctx.go:733]: @xacrimon @russjones Why not just skip over non-existent claim, just log it and not insert a placeholder? This is what we do in other places when interpolating role templates IIRC e.g. see https://github.com/gravitational/teleport/blob/v7.0.0-dev.9/lib/services/role.go#L404.

Comment (xacrimon) [reply lib/srv/ctx.go:733]: Seems better to be consistent then. Adjusted

File: lib/srv/ctx.go (hunk 1 above)
Comment (xacrimon) [reply lib/srv/ctx.go:732]: Fixed

**Review by r0mant — APPROVED** [review]

**Review by awly — APPROVED** [review]

**Pull request merged**